kDriveCore/Data/Cache/DriveFileManager/DriveFileManager+AvailableOffline.swift

```diff
@@ -21,13 +21,14 @@ import RealmSwift
 
 public extension DriveFileManager {
     func getAvailableOfflineFiles(sortType: SortType = .nameAZ) -> [File] {
-        let realm = getRealm()
-        realm.refresh()
-        let offlineFiles = realm.objects(File.self)
-            .filter(NSPredicate(format: "isAvailableOffline = true"))
-            .sorted(by: [sortType.value.sortDescriptor]).freeze()
+        let frozenFiles = fetchResults(ofType: File.self) { realm in
+            return realm.objects(File.self)
+                .filter("isAvailableOffline = true")
+                .sorted(by: [sortType.value.sortDescriptor])
+                .freeze()
+        }
 
-        return offlineFiles.map { $0.freeze() }
+        return frozenFiles.compactMap { !$0.isInvalidated ? $0 : nil }
```

Comment:
We shouldn't need to check for isInvalidated because we are using the objects we've just fetched from Realm

Reply:
(Maybe there are also other places in the modifications where you can remove it)

Reply:
I think it was added as I saw a crash, as if a deletion is probably performed on another thread it could be invalidated here, so I'm on the fence about this one.

Reply:
Everything happen serially within those calls so the objects cannot be invalid here